raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -938,6 +950,25 @@ private boolean handleFindQuorumResponse(
         }
     }
 
+    private DescribeQuorumResponseData handleDescribeQuorumRequest() {
+        if (!quorum.isLeader()) {
+            return new DescribeQuorumResponseData()
+                       .setTopics(Collections.singletonList(new DescribeQuorumTopicResponse()
+                       .setTopicName(metadataTopicPartition.topic())
```

Comment:
The handling in regard to the topic/partition seems a little curious. No matter what topic/partition is requested, we always return the metadata topic.

There is one important question we haven't fully settled on. Is the metadata topic hard-coded or do we allow it in configuration? If it's the latter, then how is a user going to discover the topic name in order to send DescribeQuorum?

One of the nice things about the group/transaction coordinator APIs is that the topic name is not part of the request. Internally we could use something other than `__consumer_offsets` and everything would be fine. I wonder if we need to rethink this API in similar terms so that we can keep the topic name internal. For example, we could change this API to something that is more focused on the metadata log, say `DescribeMetadataQuorum`. Then we could let the topic name be implicit. This might also give us a way to expose details which are specific to the metadata quorum. For example, we might want to include some information from the controller state machine.

Reply:
I think the topic name is a detail which could be hidden in the code. We just need to provide a static function like `DescribeQuorumRequest.createDescribeControllerQuorumRequest()` which user could use.

Reply:
As of current, we could still assume all the DescribeQuorum requests are looking for the metadata partition.

Reply:
I think the implementation depends on whether in multi-raft, we would let one RaftClient to own multiple ReplicatedLog or keep it 1-to-1, i.e. multiple clients with multiple logs inside the KafkaServer.

If it is the former, then we should make this agnostic to client and perhaps only encode the topic-partition information inside the ReplicatedLog.